docs/Plugins-Guide.md

```diff
@@ -271,6 +271,28 @@ module.exports = fp(dbPlugin)
 ```
 You can also tell to `fastify-plugin` to check the installed version of Fastify, in case of you need a specific api.
 
+As we mentioned earlier, Fastify starts loading its plugins __after__ `.listen()`, `.inject()` or `.ready()` are called and as such, __after__ they have been declared. This means that, even though the plugin may inject variables to the global scope via `decorate`, the decorated variables will not be accessible before calling `.listen()`, `.inject()` or `.ready()`.
+
+In case you rely on a variable injected by a preceding plugin and want to pass that in the `options` argument of `register`, you can do so by using a function instead of an object:
+```js
+const fastify = require('fastify')
+const fp = require('fastify-plugin')
+const dbClient = require('db-client')
+
+function dbPlugin (fastify, opts, next) {
+  dbClient.connect(opts.url, (err, conn) => {
+    fastify.decorate('db', conn)
+    next()
+  })
+}
+
+fastify.register(fp(dbPlugin), { url: 'https://example.com' })
+fastify.register(require('your-plugin'), parent => {
+  return { connection: parent.db, otherOption: 'foo-bar' }
+})
+```
+In the above example, the `parent` variable of the function passed in as the second argument of `register` refers to the latest state of the **global fastify instance**. This means that we are able to access any variables that were injected by preceding plugins in the order of declaration.
```

Comment:
It's not the _global_ fastify instance. It's the fastify instance that the current plugin is defined on. In your example, is `fastify`.

Reply:
You are correct, *global* refers to a very specific thing. However it is not the fastify instance that the current plugin is defined on, it is the fastify instance after the preceding plugins have been declared in the asynchronous fashion. I have reworded it a bit in the latest commit.